tests/trainer/test_dataloaders.py

```diff
@@ -1029,43 +1029,63 @@ def test_dataloaders_load_only_once_no_sanity_check(tmpdir):
         assert call['name'] == expected
 
 
-@mock.patch.dict(os.environ, {"PL_DEV_DEBUG": "1"})
-def test_dataloaders_load_every_epoch(tmpdir):
+@pytest.mark.parametrize("n", [1, 2])
+def test_dataloaders_load_every_n_epochs(tmpdir, n):
 
-    model = EvalModelTemplate()
+    model = BoringModel()
 
-    # logger file to get meta
     trainer = Trainer(
         default_root_dir=tmpdir,
         limit_train_batches=0.3,
         limit_val_batches=0.3,
-        reload_dataloaders_every_epoch=True,
+        reload_dataloaders_every_n_epochs=n,
         max_epochs=3,
     )
-    result = trainer.fit(model)
-
+    trainer.fit(model)
     trainer.test()
 
-    assert len(trainer.dev_debugger.val_dataloader_calls) == 4
-    assert len(trainer.dev_debugger.train_dataloader_calls) == 3
-    assert len(trainer.dev_debugger.test_dataloader_calls) == 1
-
     # verify the sequence
     calls = trainer.dev_debugger.dataloader_sequence_calls
-    expected_sequence = [
-        'val_dataloader',
-        'train_dataloader',
-        'val_dataloader',
-        'train_dataloader',
-        'val_dataloader',
-        'train_dataloader',
-        'val_dataloader',
-        'test_dataloader'
-    ]
+    if n == 1:
+        expected_sequence = [
+            'val_dataloader',
+            'train_dataloader',
+            'val_dataloader',
+            'train_dataloader',
+            'val_dataloader',
+            'train_dataloader',
+            'val_dataloader',
+            'test_dataloader'
+        ]
+    elif n == 2:
+        expected_sequence = [
+            'val_dataloader',
+            'train_dataloader',
+            'val_dataloader',
+            'train_dataloader',
+            'val_dataloader',
+            'test_dataloader'
+        ]
+
     for call, expected in zip(calls, expected_sequence):
         assert call['name'] == expected
 
 
+@pytest.mark.parametrize("n", ['test', -1])
+def test_dataloaders_load_every_n_epochs_exception(tmpdir, n):
+
+    with pytest.raises(Exception):
```

Comment:
Catch the actual exception (`MisconfigurationException`, I think), also `match` the text

And put this context manager where the exception is raised (just the Trainer creation)